service/history/task/cross_cluster_source_task_executor.go

```diff
@@ -22,30 +22,410 @@
+	} else if err == errContinueExecution {
+		err = nil
+	}
+
+	return err
+}
+
+func (t *crossClusterSourceTaskExecutor) executeStartChildExecutionTask(
+	ctx context.Context,
+	task *crossClusterSourceTask,
+) (retError error) {
+
+	taskInfo := task.GetInfo().(*persistence.CrossClusterTaskInfo)
+	wfContext, mutableState, release, err := loadWorkflowForCrossClusterTask(ctx, t.executionCache, taskInfo, t.metricsClient, t.logger)
+	if err != nil || mutableState == nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	initiatedEventID := taskInfo.ScheduleID
+	childInfo, ok := mutableState.GetChildExecutionInfo(initiatedEventID)
+	if !ok {
+		return nil
+	}
+
+	ok, err = verifyTaskVersion(t.shard, t.logger, taskInfo.DomainID, childInfo.Version, taskInfo.Version, task)
+	if err != nil || !ok {
+		return err
+	}
+
+	initiatedEvent, err := mutableState.GetChildExecutionInitiatedEvent(ctx, initiatedEventID)
+	if err != nil {
+		return err
+	}
+
+	if task.ProcessingState() == processingStateInvalidated {
+		return t.generateNewTask(ctx, wfContext, mutableState, task)
+	}
+
+	// handle common errors
+	failedCause := task.response.FailedCause
+	if failedCause != nil {
+		switch *failedCause {
+		case types.CrossClusterTaskFailedCauseDomainNotActive:
+			return t.generateNewTask(ctx, wfContext, mutableState, task)
+		case types.CrossClusterTaskFailedCauseWorkflowNotExists:
+			return &types.EntityNotExistsError{}
+		case types.CrossClusterTaskFailedCauseWorkflowAlreadyCompleted:
+			return &types.WorkflowExecutionAlreadyCompletedError{}
+		case types.CrossClusterTaskFailedCauseUncategorized:
+			// ideally this case should not happen
+			// crossClusterSourceTask.Update() will reject response with this failed cause
+			t.setTaskState(task, ctask.TaskStatePending, processingState(task.response.TaskState))
+			return errContinueExecution
+		}
+	}
+
+	switch processingState(task.response.TaskState) {
+	case processingStateInitialized:
+		if childInfo.StartedID != common.EmptyEventID {
+			t.setTaskState(task, ctask.TaskStatePending, processingStateResponseRecorded)
+			return errContinueExecution
+		}
+
+		attributes := initiatedEvent.StartChildWorkflowExecutionInitiatedEventAttributes
+		now := t.shard.GetTimeSource().Now()
+		if failedCause != nil &&
+			(*failedCause == types.CrossClusterTaskFailedCauseDomainNotExists ||
+				*failedCause == types.CrossClusterTaskFailedCauseWorkflowAlreadyRunning) {
+			return recordStartChildExecutionFailed(ctx, taskInfo, wfContext, attributes, now)
+		}
+
+		childRunID := task.response.StartChildExecutionAttributes.GetRunID()
+		err := recordChildExecutionStarted(ctx, taskInfo, wfContext, attributes, childRunID, now)
+		if err != nil {
+			return err
+		}
+
+		// advance the task state so that it will be available for polling again
+		t.setTaskState(task, ctask.TaskStatePending, processingStateResponseRecorded)
+		return errContinueExecution
```

Comment:
return error here to prevent task from acked?

Reply:
`errContinueExecution` is returned here and then ignored on L107. The whole Execute() don't have to return an error and it's ok to call Ack() method on the task since Ack() method won't actually set the task state to acked, so the task is still pending and available for polling.